sql/src/main/java/io/crate/operation/reference/sys/node/NodeStatsContextFieldResolver.java

```diff
@@ -0,0 +1,244 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.reference.sys.node;
+
+import com.google.common.collect.ImmutableMap;
+import io.crate.Build;
+import io.crate.Version;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.ReferenceIdent;
+import io.crate.metadata.sys.SysNodesTableInfo;
+import io.crate.monitor.ExtendedNodeInfo;
+import io.crate.monitor.ThreadPools;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.admin.cluster.node.info.NodeInfo;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.transport.TransportAddress;
+import org.elasticsearch.common.util.Consumer;
+import org.elasticsearch.monitor.jvm.JvmService;
+import org.elasticsearch.monitor.os.OsService;
+import org.elasticsearch.node.service.NodeService;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+
+@Singleton
+public class NodeStatsContextFieldResolver {
+
+    private final static ESLogger LOGGER = Loggers.getLogger(NodeStatsContextFieldResolver.class);
+
+    private final OsService osService;
+    private final JvmService jvmService;
+    private final ClusterService clusterService;
+    private final NodeService nodeService;
+    private final ThreadPool threadPool;
+    private final ExtendedNodeInfo extendedNodeInfo;
+
+    @Inject
+    public NodeStatsContextFieldResolver(ClusterService clusterService,
+                                         OsService osService,
+                                         NodeService nodeService,
+                                         JvmService jvmService,
+                                         ThreadPool threadPool,
+                                         ExtendedNodeInfo extendedNodeInfo) {
+        this.osService = osService;
+        this.jvmService = jvmService;
+        this.clusterService = clusterService;
+        this.nodeService = nodeService;
+        this.threadPool = threadPool;
+        this.extendedNodeInfo = extendedNodeInfo;
+    }
+
+    public NodeStatsContext forColumns(Collection<ReferenceIdent> columns) {
```

Comment:
That's the point - I don't think that tableIdent is required here - This whole Resolver is already specific to sys.nodes. There is no need to pretend that it can handle other tableIdents